src/app/mesh/qgsmeshcalculatordialog.cpp

```diff
@@ -45,14 +46,18 @@ QgsMeshCalculatorDialog::QgsMeshCalculatorDialog( QgsMeshLayer *meshLayer, QWidg
   QgsGui::enableAutoGeometryRestore( this );
 
   cboLayerMask->setFilters( QgsMapLayerProxyModel::PolygonLayer );
-  getDatasetGroupNames();
+  QgsMeshDatasetGroupListModel *model = new QgsMeshDatasetGroupListModel( this );
```

Comment:
std::unique_ptr< QgsMeshDatasetGroupListModel>

Reply:
Model is a QObject and it has QgsMeshCalculatorDialog ("this") as parent and will be destruct with it

Reply:
that I understand,  but with solution I suggested there will be no memory leak never (even if the c++ code throw some exception on line 50) and with .release() you transfer ownership